feature-libs/cart/base/root/events/multi-cart-event.listener.ts

```diff
@@ -23,7 +23,9 @@ export class MultiCartEventListener implements OnDestroy {
   protected onCartBaseAction(): void {
     this.subscriptions.add(
       this.eventService.get(LoadCartEvent).subscribe(({ userId, cartId }) => {
-        this.multiCartFacade.loadCart({ userId, cartId });
+        if (userId && cartId) {
```

Comment:
reason why this needs to be done was from the previous checkout event refactoring https://github.com/SAP/spartacus/pull/15789

We now dispatch an extra layer of event to help with readability + tracing, but now it causes issues, where some events requires additional payloads.

------
This is really not the issue to fix in guest checkout. This fixes the error we get after login in, where it says "lack of permissions". This changes in checkout-event.module.ts was the truthy fix as it was forgotten to bootstrap the event listener

Reply:
for the reviewiwer:

OPTION 1: conditional userId and cartId to dispatch


OR

OPTION 2: should I change the logic for the LoginEvent to accept the userId and cartId payload? This will remove this conditional statement. I believe we can go this route as it make sense that the user want to know the current userId and cartId of the logged in user. What do you think?